src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,99 @@
+    private static final String CANNOT_FIND_USER_BY_ID = "Can't find user by id: ";
+    private static final String CANNOT_FIND_ORDER_USER_ID = "Can't find order by user id: ";
+    private static final String CANNOT_FIND_SHOPPING_CART_BY_USER_ID
+            = "Can't find shopping cart by user id: ";
+    private final OrderRepository orderRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final UserRepository userRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final CartItemRepository cartItemRepository;
+    private final OrderMapper orderMapper;
+
+    @Override
+    @Transactional
+    public OrderResponseDto placeOrder(AddressRequestDto requestDto, Long userId) {
+        ShoppingCart shoppingCart = shoppingCartRepository.getShoppingCartByUserId(userId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        CANNOT_FIND_SHOPPING_CART_BY_USER_ID + userId));
+
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new EntityNotFoundException(CANNOT_FIND_USER_BY_ID + userId));
+
+        Order order = new Order();
+
+        BigDecimal total = shoppingCart.getCartItems().stream()
+                .map(item -> item.getBook().getPrice()
+                        .multiply(BigDecimal.valueOf(item.getQuantity())))
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+
+        order.setTotal(total);
+        order.setUser(user);
+        order.setShippingAddress(requestDto.shippingAddress());
+        order.setStatus(Order.Status.PENDING);
+        order = orderRepository.save(order);
+
+        Set<OrderItem> orderItems = new HashSet<>();
+        for (CartItem cartItem : shoppingCart.getCartItems()) {
+            OrderItem orderItem = new OrderItem();
+
+            orderItem.setOrder(order);
+            orderItem.setBook(cartItem.getBook());
+            orderItem.setPrice(cartItem.getBook().getPrice());
+            orderItem.setQuantity(cartItem.getQuantity());
+            orderItemRepository.save(orderItem);
+            orderItems.add(orderItem);
+        }
+
+        order.setOrderItems(orderItems);
+        order.setOrderDate(LocalDateTime.now());
+        cartItemRepository.deleteAll(shoppingCart.getCartItems());
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    public List<OrderResponseDto> getUserOrders(Long userId) {
+        List<Order> orders = orderRepository.findOrdersByUserId(userId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        CANNOT_FIND_ORDER_USER_ID + userId));
+        return orders.stream().map(orderMapper::toDto).collect(Collectors.toList());
+    }
+
+    @Override
+    public OrderResponseDto updateOrderStatus(Long orderId, OrderStatusRequestDto requestDto) {
+        Order orderById = orderRepository.findOrderById(orderId);
+        orderById.setStatus(Order.Status.valueOf(requestDto.getStatus()));
```

Comment:
If will change the field type in OrderStatusRequestDto from String to Order.Status can be used:
```suggestion
        orderById.setStatus(requestDto.getStatus());
```